src/Npgsql/TypeMapping/ConnectorTypeMapper.cs

```diff
@@ -380,20 +380,12 @@ void BindRangeType(NpgsqlTypeHandler elementHandler, PostgresRangeType pgRangeTy
                 ? NpgsqlDbType.Range | elementNpgsqlDbType.Value
                 : (NpgsqlDbType?)null;
 
+            // We only want to bind supported range CLR types whose element CLR types are being bound as well.
+            var clrTypes = rangeHandler.SupportedElementClrTypes
+                .Where(t => elementClrTypes.Contains(t.GenericTypeArguments[0]))
+                .ToList();
```

Comment:
And this one too. You produce a list of supported types (two allocations, one for list and another one for an internal buffer). In line 388 you are doing `ToArray` and introduce the third.

Reply:
This is initialization code that happens once per physical connection - the impact of allocations here is non-existent. It's very important to distinguish between hot paths and very cold paths, where it's fine to use LINQ and other relatively inefficient code-improving techniques...

Reply:
On an unrelated note, if we think there's a perf issue with physical connection setup time (which is only an app setup time), we could consider initializing the whole type handler system lazily rather than eagerly, as we do now. However, we haven't seen any evidence (or gotten complaints) that this is the case, and overheads such as networking, auth, PostgreSQL-side process spinup etc. are very likely to make this completely negligible.